src/NuGet.Core/NuGet.Protocol.Core.v3/LocalRepositories/FindLocalPackagesResourceV3.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// Retrieve packages from a local folder or UNC share that uses the V3 folder structure.
+    /// </summary>
+    public class FindLocalPackagesResourceV3 : FindLocalPackagesResource
+    {
+        public FindLocalPackagesResourceV3(string root)
+        {
+            Root = root;
+        }
+
+        public override IEnumerable<LocalPackageInfo> FindPackagesById(string id, ILogger logger, CancellationToken token)
+        {
+            token.ThrowIfCancellationRequested();
```

Comment:
Async or not we can still cancel to avoid unneeded work. This was the pattern used before for the local resources.